### ISSUE TITLE
OSX Build and Deployment problems

### ISSUE THREAD

**Issue by johnc44** [awslabs/aws-sam-cli#1063]
On Mac OSX Mojave, sam build and sam package don't work for me.
```
sam build
2019-03-14 09:13:58 Building resource 'MyLambda'
2019-03-14 09:13:58 Running PythonPipBuilder:ResolveDependencies
Build Failed
Error: PythonPipBuilder:ResolveDependencies - {cffi==1.12.2(wheel), pyyaml==5.1(sdist)}
```

Adding --use-container works, however.

Those dependencies do exist.  I've tried uninstalling and reinstalling them.

```
sam package --output-template-file .aws-sam/deploy/my-app.yml --s3-bucket my-lambda-repository
Error: Unable to find AWS CLI installation under following names: ['aws']
make: *** [deploy] Error 1
```

Using aws cloudformation instead of sam works.

I remember when I tried to install sam it wouldn't install through brew and I had to use pip.  (Cant remember what the exact problem was now.).  I also have to activate a venv beforehand otherwise sam can't find python.  (On OSX 'python' is 2.7, but I have an alias to point it to 3.7 in my .bashprofile)

**Comment by jfuss** [comment]
@johnc44 Looks like you might have some native bindings for your function. Which would mean `--use-container` is what you should do to build. 

On the package part, it looks like SAM CLI can't find your installation of AWS CLI. This is mostly likely due to how you have installed. I recommend uninstalling with pipe and venv, and instead go to brew. By the sounds of it, you might have run into [this issue](https://github.com/aws/homebrew-tap/issues/6) with brew. If you find it's not, feel free to cut us an issue on that or reach out in the #samdev channel on slack.

**Comment by johnc44** [comment]
Thanks for the reply - I'll have a look at the issue you mention over the weekend and see if I can get brew to work.

We do have a few problems with "sam build" vs "sam build --use-container".  I've been meaning to raise them on here but haven't; had a chance to gather enough info.  But briefly:

* "sam build" on OSX doesn't seem to work (as above)
* "sam build --use-container" doesn't work if the repo has symlinks.
* "sam build --use-container" doesn't seem to work in Ubuntu WSL.  (Although I can use sam local to test the API from here)  I get Error: PythonPipBuilder:ResolveDependencies - Requirements file not found: /tmp/samcli/source/requirements.txt
* "sam build" doesn't work in Ubuntu WSL if using a Windows drive (i.e. if I'm in /mnt/c/....).  But if I copy/git clone the code elsewhere (e.g. /home/me...) then it's fine.  Error it gives is Error: PythonPipBuilder:CopySource - name 'WindowsError' is not defined
* Obviously some dependencies may require "--use-container" to work, although we currently don't use any that I'm aware of.

We have one project containing multiple lambdas that use some shared code currently imported in using symlinks.  Although considering using git links in requirements.txt to bring them in at some point - although I'm not sure that's going to work well with --use-container either.

We're trying to define a makefile that allows us to build regardless of platform, but haven't yet managed to get something that works for everyone.

**Comment by johnc44** [comment]
I've fixed the problem... I found that I had awscli installed in 3 different places.  Once I removed it from everywhere and reinstalled it with the package installer, then everything started to work.  Even "sam build" works now.

One weird thing... and not sure if this is a bash thing or an OSX thing, but I was doing "which aws" and getting nothing back, but "aws --version" still showed up a version.  I found it by "command -v aws".  (This was after I'd found 2 instances which did come back from which).

Also my version of the CLI was 1.15.x not 1.16.x so that might have something to do with it perhaps.
